### ISSUE TITLE
[MPS] torch.matmul produces transposed multiplication result

### ISSUE THREAD

**Issue by mh105** [pytorch/pytorch#86937]
### 🐛 Describe the bug

As of PyTorch stable 1.12.1, doing matrix multiplication of tensors on MPS devices produces the incorrect transposed result. 

```
A = torch.as_tensor([[1.,2.], [3.,4.]], dtype=torch.float32, device='mps')
A
```

> Out[52]: 
> tensor([[1.0000, 2.0000],
>         [3.0000, 4.0000]], device='mps:0')

        
`torch.matmul(A, A)`

> Out[53]: 
> tensor([[ 5.0000, 11.0000],
>         [11.0000, 25.0000]], device='mps:0')

`torch.matmul(A, A.T)`

> Out[54]: 
> tensor([[ 7.0000, 10.0000],
>         [15.0000, 22.0000]], device='mps:0')

The opposite should be true. 

### Versions

PyTorch version: 1.12.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6 (arm64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.102)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:00:33)  [Clang 13.0.1 ] (64-bit runtime)
Python platform: macOS-12.6-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.3
[pip3] pytest-pytorch==0.0.0
[pip3] torch==1.12.1
[conda] numpy                     1.23.3           py39hcb4b507_0    conda-forge
[conda] pytest-pytorch            0.2.1              pyh44b312d_0    conda-forge
[conda] pytorch                   1.12.1                  py3.9_0    pytorch

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

**Comment by malfet** [comment]
This has already been fixed in latest trunk and in 1.13.0 release candidate

**Comment by malfet** [comment]
Duplicate issue https://github.com/pytorch/pytorch/issues/85406, fix is in https://github.com/pytorch/pytorch/pull/83124
